dotCMS/src/main/java/com/dotmarketing/webdav/FolderResourceImpl.java

```diff
@@ -72,20 +74,23 @@ public CollectionResource createCollection(String newName) throws DotRuntimeExce
 			} catch (DotSecurityException e) {
 				Logger.error(DotWebdavHelper.class, e.getMessage(), e);
 				throw new DotRuntimeException(e.getMessage(), e);
-			}			
-			
-			dotDavHelper.createTempFolder(File.separator + host.getHostname() + folderPath + File.separator + newName);
-			File f = new File(File.separator + host.getHostname() + folderPath);
-			TempFolderResourceImpl tr = new TempFolderResourceImpl(f.getPath(),f ,isAutoPub);
-			return tr;
+			}
+
+            final String hostFolderPath = new StringBuilder(File.separator).append(host.getHostname())
+					.append(!folderPath.endsWith(File.separator)?folderPath + File.separator : folderPath).toString();
+
+            dotDavHelper.createTempFolder(hostFolderPath + newName);
+			File file = new File(File.separator + host.getHostname() + folderPath);
+			TempFolderResourceImpl tempFolderResource = new TempFolderResourceImpl(file.getPath(),file ,isAutoPub);
+			return tempFolderResource;
 		}
 		if(!path.endsWith("/")){
 			path = path + "/";
 		}
 		try {
-			Folder f = dotDavHelper.createFolder(path + newName, user);
-			FolderResourceImpl fr = new FolderResourceImpl(f, path + newName + "/");
-			return fr;
+			Folder folder = dotDavHelper.createFolder(path + newName, user);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Rename "folder" which hides the field declared at line 47. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AHiddenFieldCheck)